src/mlpack/methods/reinforcement_learning/q_learning.hpp

```diff
@@ -0,0 +1,162 @@
+/**
+ * @file q_learning.hpp
+ * @author Shangtong Zhang
+ *
+ * This file is the definition of QLearning class,
+ * which implements Q-Learning algorithms.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#ifndef MLPACK_METHODS_RL_Q_LEARNING_HPP
+#define MLPACK_METHODS_RL_Q_LEARNING_HPP
+
+#include <mlpack/prereqs.hpp>
+
+#include "replay/random_replay.hpp"
+
+namespace mlpack {
+namespace rl {
+
+/**
+ * Implementation of various Q-Learning algorithms, such as DQN, double DQN.
+ *
+ * For more details, see the following:
```

Comment:
What do you think about:

```bibtex
@article{Mnih2013,
  author    = {Volodymyr Mnih and
               Koray Kavukcuoglu and
               David Silver and
               Alex Graves and
               Ioannis Antonoglou and
               Daan Wierstra and
               Martin A. Riedmiller},
  title     = {Playing Atari with Deep Reinforcement Learning},
  journal   = {CoRR},
  year      = {2013},
  url       = {http://arxiv.org/abs/1312.5602}
}
```

it's not that short, but I think this will work just fine.